pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -435,6 +455,116 @@ static void handleQueryLimitOverride(BrokerRequest brokerRequest, int queryLimit
     }
   }
 
+  /**
+   * Check if a SQL parsed BrokerRequest is a literal only query.
+   * @param brokerRequest
+   * @return true if this query selects only Literals
+   *
+   */
+  @VisibleForTesting
+  static boolean isLiteralOnlyQuery(BrokerRequest brokerRequest) {
+    if (brokerRequest.getPinotQuery() != null) {
```

Comment:
Would be good to state in the PR description that this feature is only supported for SQL mode.

Reply:
sure